lib/tds.js

```diff
@@ -324,14 +355,16 @@ async function generateDNRRulesForTrackerEntry (
         }
     }
 
-    return removeRedundantDNRRules(dnrRules)
+    const trackerRules = removeRedundantDNRRules(dnrRules)
+    return { trackerRules }
 }
 
 function finalizeDNRRulesAndLookup (startingRuleId, dnrRules) {
     const ruleIdByByStringifiedDNRRule = new Map()
     const requestDomainsByRuleId = new Map()
     const trackerDomainsByRuleId = new Map()
     const matchDetailsByRuleId = {}
+    const inverseCustomActionRules = []
```

Comment:
Should this not be an Object that maps each customAction its array of inverse rules?
```suggestion
    const inverseRulesByCustomAction = {}
```

Reply:
(This is covered in my above comment, resolving.)